### PR TITLE
migrate from cJSON to zcbor

### PR DESCRIPTION
**Pull request by mniestroj**
~Depends on: #123~
~Depends on: #124~
Depends on: #126

TODO:
- [x] convert unit tests from cJSON to zcbor

### CODE REVIEW

**Comment by mniestroj** [comment]
Unit tests of RPC need to be rewritten, but other than that, this PR is ready to be reviewed.

**Review by sam-golioth — CHANGES_REQUESTED** [review]

File: port/esp_idf/components/golioth_sdk/CMakeLists.txt [hunk 1]
```diff
@@ -47,6 +48,7 @@ idf_component_register(
         "${sdk_src}/golioth_event_group.c"
         "${sdk_src}/golioth_mbox.c"
         "${sdk_src}/golioth_fw_block_processor.c"
+        "${sdk_src}/zcbor_utils.c"
```
[the remark this answers is not in the document]
Comment (mniestroj) [reply port/esp_idf/components/golioth_sdk/CMakeLists.txt:51]: Moved to the commit which introduces this file.

File: src/zcbor_utils.h [hunk 2]
```diff
@@ -0,0 +1,71 @@
...
+    }
+
+    return false;
+}
+
+int zcbor_map_int64_decode(zcbor_state_t* zsd, void* value);
+
+int zcbor_map_tstr_decode(zcbor_state_t* zsd, void* value);
+
+int zcbor_map_decode(zcbor_state_t* zsd, struct zcbor_map_entry* entries, size_t num_entries);
```
[the remark this answers is not in the document]
Comment (mniestroj) [reply src/zcbor_utils.h:82]: Sure, added doxygen comments.

File: src/zcbor_utils.h [hunk 3]
```diff
@@ -0,0 +1,71 @@
...
+};
+
+static inline bool zcbor_list_or_map_end(zcbor_state_t* state) {
+    if (state->indefinite_length_array) {
+        return *state->payload == 0xff;
+    } else {
+        return state->elem_count == 0;
+    }
+
+    return false;
```
[the remark this answers is not in the document]
Comment (mniestroj) [reply src/zcbor_utils.h]: Good call, simplified that function by removing `else` case.

**Review by sam-golioth — APPROVED** [review]
The RPC tests are in the remote shell commit, but other than that looks great!

**Comment by mniestroj** [comment]
> The RPC tests are in the remote shell commit, but other than that looks great!

I've squashed those two, as `remote_shell` uses `rpc`, so both need to be changed at the same time.

**Pull request merged**